### PR TITLE
feat: set tcp proxy listener buffer

### PR DESCRIPTION
**Pull request by Xunzhuo**
Part of: #1048

### CODE REVIEW

File: internal/xds/translator/listener.go [hunk 1]
```diff
@@ -39,6 +40,7 @@ func buildXdsTCPListener(name, address string, port uint32) *listenerv3.Listener
 				Filter:     listenerAccessLogFilter,
 			},
 		},
+		PerConnectionBufferLimitBytes: wrapperspb.UInt32(32768),
```
Comment (zirain) [inline internal/xds/translator/listener.go:43]: should EG support to custom this?

Comment (arkodg) [reply internal/xds/translator/listener.go:43]: lets raise a issue for this ? #24 is always a back up option

Comment (arkodg) [inline internal/xds/translator/listener.go:43]: would be great if this is a magic number const, with a link to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/listener/v3/listener.proto#envoy-v3-api-field-config-listener-v3-listener-per-connection-buffer-limit-bytes

Comment (Xunzhuo) [reply internal/xds/translator/listener.go]: If so, we need to add new apis in envoyproxy too, let us keep it in this way, unless someone requests for it.

Comment (Xunzhuo) [reply internal/xds/translator/listener.go]: Done

**Review by arkodg — APPROVED** [review]
🚀

**Review by zirain — APPROVED** [review]

**Comment by haorenfsa** [comment]
Em... I just tried on v1.1.0, seems not work for me. I add a `BackendTrafficPolicy` but the `per_connection_buffer_limit_bytes` is still 32KB

**Comment by arkodg** [comment]
> Add a knob to make `per_connection_buffer_limit_bytes` configurable #2600

cc @yaelSchechter

**Pull request merged**